test/hack/cleanup/main.go

```diff
@@ -54,13 +56,18 @@ type CleanableResourceType interface {
 	Type() string
 	Get(context.Context, time.Time) ([]string, error)
 	Cleanup(context.Context, []string) ([]string, error)
+	DeleteByClusterName(context.Context, string) error
```

Comment:
Is it worth scoping our existing `Get()` logic to take in an option that would let you pass in the cluster name? If that's passed in for each of these test cases, add a filter? Alternatively, you could do a `GetByName()` call that got the ids and then call the same Cleanup() call for both sets of logic